transactor/src/main/scala/io/mediachain/copycat/TransactorService.scala

```diff
@@ -0,0 +1,132 @@
+      }
+
+    client
+      .lookup(ref)
+      .map { (x: Option[Reference]) =>
+        x.map { ref =>
+          refToRPCMultihashRef(ref)
+        }.getOrElse {
+          throw new StatusRuntimeException(Status.NOT_FOUND)
+        }
+      }
+  }
+
+  private def refToRPCMultihashRef(ref: Reference)
+  : Transactor.MultihashReference = ref match {
+    case MultihashReference(multihash) =>
+      Transactor.MultihashReference(multihash.base58)
+    case r =>
+      throw new ClassCastException(
+        s"Expected MultihashReference, got type ${r.getClass.getTypeName}"
+      )
+  }
+
+  override def insertCanonicalRecord(request: InsertRequest)
+  : Future[Transactor.MultihashReference] = {
+    val recXor = CborSerialization.fromCborBytes[CanonicalRecord](
+      request.canonicalCbor.toByteArray
+    )
+
+    val insertF = recXor match {
+      case Xor.Left(err) => throw new StatusRuntimeException(
+        Status.INVALID_ARGUMENT.withDescription(
+          s"Object deserialization error: ${err.message}"
+        )
+      )
+      case Xor.Right(obj) =>
+        client.insert(obj)
+    }
+
+    insertF.map { entryXor: Xor[JournalError, CanonicalEntry] =>
+      entryXor match {
+        case Xor.Left(err) =>
+          throw new StatusRuntimeException(
+            Status.FAILED_PRECONDITION.withDescription(
+              s"Journal Error: $err"
+            )
+          )
+        case Xor.Right(entry) =>
+          refToRPCMultihashRef(entry.ref)
+      }
+    }
+  }
+
+  override def updateCanonicalRecord(request: UpdateRequest)
+  : Future[Transactor.MultihashReference] = {
+    val cellXor = CborSerialization.fromCborBytes[ChainCell](
+      request.chainCellCbor.toByteArray
+    )
+
+    val updateF = cellXor match {
+      case Xor.Left(err) => throw new StatusRuntimeException(
+        Status.INVALID_ARGUMENT.withDescription(
+          s"Object deserialization error: ${err.message}"
+        )
+      )
+      case Xor.Right(cell) =>
+        client.update(cell.ref, cell)
+    }
+
+    updateF.map { entryXor: Xor[JournalError, ChainEntry] =>
+      entryXor match {
+        case Xor.Left(err) =>
+          throw new StatusRuntimeException(
+            Status.FAILED_PRECONDITION.withDescription(
+              s"Journal Error: $err"
+            )
+          )
+        case Xor.Right(entry) =>
+          refToRPCMultihashRef(entry.chain)
+      }
+    }
+  }
+}
+
+object TransactorService {
+  def createServerThread(service: TransactorService,
+                         executor: ExecutorService,
+                         port: Int)
+                        (implicit executionContext: ExecutionContext)
+  : Unit = {
+    import scala.language.existentials
+
+    val builder = ServerBuilder.forPort(port)
+    val server = builder.addService(
+      TransactorServiceGrpc.bindService(service, executionContext)
+    ).build
+
+    executor.submit(new Runnable {
+      override def run(): Unit = {
```

Comment:
uhm can you write this as:
```
def run {
 ...
}
```

I don't see the need for all that type noise here.